frost-core/src/batch.rs

```diff
@@ -115,49 +110,49 @@ where
         let mut VKs = Vec::with_capacity(n);
         let mut R_coeffs = Vec::with_capacity(self.signatures.len());
         let mut Rs = Vec::with_capacity(self.signatures.len());
-        let mut P_coeff_acc = Scalar::zero();
+        let mut P_coeff_acc = <<C::Group as Group>::Field as Field>::zero();
 
         for item in self.signatures.iter() {
-            let (z_bytes, R_bytes, c) = (item.sig.z_bytes, item.sig.R_bytes, item.c);
-
-            let s = Scalar::from_bytes_mod_order(z_bytes);
-
-            let R = {
-                match CompressedRistretto::from_slice(&R_bytes).decompress() {
-                    Some(point) => point,
-                    None => return Err(Error::InvalidSignature),
-                }
-            };
+            let z = item.sig.z;
+            let R = item.sig.R;
 
-            let VK = VerifyingKey::try_from(item.vk_bytes.bytes)?.point;
+            let blind = <<C::Group as Group>::Field as Field>::random(&mut rng);
 
-            let z = Scalar::random(&mut rng);
+            let P_coeff = blind * z;
+            P_coeff_acc = P_coeff_acc - P_coeff;
 
-            let P_coeff = z * s;
-            P_coeff_acc -= P_coeff;
-
-            R_coeffs.push(z);
+            R_coeffs.push(blind);
             Rs.push(R);
 
-            VK_coeffs.push(Scalar::zero() + (z * c));
-            VKs.push(VK);
+            VK_coeffs.push(<<C::Group as Group>::Field as Field>::zero() + (z * item.c.0));
```

Comment:
As discussed in chat

```suggestion
            VK_coeffs.push(<<C::Group as Group>::Field as Field>::zero() + (blind * item.c.0));
```